### ISSUE TITLE
Transform `var` to `let` or `const` if possible, warn if not

### ISSUE THREAD

**Issue by eventualbuddha** [esnext/esnext#76]
```js
console.log(zero);
var zero = 0;
var i = 1;
i++;
var PI = 3.14;

// ↑ becomes ↓

// WARNING: uninitialized-var  variable `zero` used before it is initialized
console.log(zero);
var zero = 0;
let i = 1;
i++;
const PI = 3.14;
```

**Comment by thejameskyle** [comment]
Also:

```js
for (var i = 0; i < arr.length; i++) {
  // ...
}
console.log(i);
```

**Comment by eventualbuddha** [comment]
@thejameskyle ah, yeah. That's an annoying one.

**Comment by eventualbuddha** [comment]
Added in f43bdd7d6b13183ec4dc5c7577b22f906c0681b2, released in v1.2.0.
